modules/standard/Random.chpl

```diff
@@ -576,20 +733,38 @@ module Random {
       /*
         Returns the next value in the random stream.
 
-        Reals are between 0 and 1; integers are in the range of the integer.
-        :returns: The next value in the random stream as type :type:`eltType`.
+        Generated reals are in [0,1] - both 0.0 and 1.0 are
+        possible values.
+
+        Generated integers cover the full value range of the integer.
+
+        :arg resultType: the type of the result. Defaults to :type:`eltType`.
+          `resultType` must be the same or a smaller size number.
+        :returns: The next value in the random stream as type :type:`resultType`.
        */
-      proc getNext(): eltType {
+      proc getNext(type resultType=eltType): resultType {
```

Comment:
This is one of @ben-albrecht's feature requests. It seemed easy enough to implement to me. Note though that I didn't make the RandomStreamInterface do this, so it is PCGRandomStream specific.

The reason I think it makes sense is that the RNG (like all RNGs) produces a bunch of random bits. Getting different types of numbers out is just different uses of those bits (except for some issues about how many bits you get in each call, which I handled here with a compiler error, but which I could also handle transparently in a way similar to how I implemented the bounded integers).

However, I don't have a strong opinion about this and it would be relatively easy to remove it this week if we decide to. One interesting alternative is to require that the user create another RandomStream with the same seed and then advance it to the same position.

This is one of those features that might make *serial* uses of Random more reasonable. However, it might make more sense for a serial RandomStream to be a totally different interface.